server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -120,7 +120,7 @@ protected void masterOperation(final RolloverRequest rolloverRequest, final Clus
         final String rolloverIndexName = indexNameExpressionResolver.resolveDateMathExpression(unresolvedName);
         MetaDataCreateIndexService.validateIndexName(rolloverIndexName, state); // will fail if the index already exists
         checkNoDuplicatedAliasInIndexTemplate(metaData, rolloverIndexName, rolloverRequest.getAlias());
-        client.admin().indices().prepareStats(sourceIndexName).clear().setDocs(true).execute(
+        client.admin().indices().prepareStats(rolloverRequest.getAlias()).clear().setDocs(true).execute(
```

Comment:
I will have to check, but I'm not sure this is the same thing.

will stats for shards of all the matched read-aliases be returned in this case? For example,
If the alias points to multiple indices, then we may be checking stats of indices that are of no 
concern to us.

Reply:
update: this does have semantic differences given how the stats response is interpreted. Looks like a solution to correct this exists (in separate review comment)

Reply:
That is a good question. Although it is unfortunate that we are collecting stats on potentially a lot of indices/shards, I am not sure performance is of the upmost importance for this API. This API is a management api that already takes a while due to allocation lag, etc. 

It is good to note that this stats call would potentially hit hundreds of indices, assuming a user has daily rollover indices with ILM, but the API is designed to handle this in general, so it feels safe to do this internally here.